client/auth.go

```diff
@@ -292,6 +309,11 @@ func (c *Conn) writeAuthHandshake() error {
 
 	// Filler [23 bytes] (all 0x00)
 	pos := 13
+	if collation.ID > 255 {
+		// skip setting the first byte of the filler to 0x00 since it is used to
+		// send the upper 8 bits of the collation id
+		pos++
+	}
 	for ; pos < 13+23; pos++ {
```

Comment:
Is the 23 here correct? Or should it be 22 if the collation is 2 bytes?

Reply:
> Is the 23 here correct? Or should it be 22 if the collation is 2 bytes?

yes you are correct, should be 22